### DISCUSSION TITLE
Expand not working on relationship fields that are type Array

### DISCUSSION THREAD

**Discussion by RA9** [pocketbase/pocketbase#1845] [Q&A]
I have a relationship field that is an array but it doesn't show the expand properties each time I make a query. Is there a way to do it if you have a relationship field that is a type of Array?

**Comment by arslanastral** [comment]
Can you share the code ? Are you actually expanding it in the query ? like this:

```ts 
await pb.collection('my_collection').getOne(<id>, { expand: <relation_field> } )
```
If you are, can you check if `<relation_field>` collection [API Rules](https://pocketbase.io/docs/api-rules-and-filters) are set properly and allow you to perform list/search or view actions.

**Comment by Alexilator** [comment]
If I could chime in here. I am having the same issue I think and I have documented it here. First and foremost: all API Rules for both collections `testrel1`and `testrel2` are set to be fully open, no custom rules applied.
My request: On `testrel2` using `expand` on field `rel2`, the relation field. Below that, I do a direct request on the entry `test1` inside `testrel1`, which is also the field linked inside the relation `rel2`
```ts
const record = await pb.collection('testrel2').getOne('vfjjmyu7dn1cm7u',{

expand: 'rel2'

})
const record2 = await pb.collection('testrel1').getOne('wu7sqtqzbz48z2g')

console.log(record)
console.log(record2)
```

**output with single relation:**
```js
// record
{
  collectionId: 'p6fvms0nkje2n2t',
  collectionName: 'testrel2',
  created: '2023-09-06 12:48:16.962Z',
  expand: {
    rel2: {
      collectionId: 'jdr3sz2uro70m1i',
      collectionName: 'testrel1',
      created: '2023-09-06 12:48:23.308Z',
      id: '5hrhujcfz6k6qon',
      text: 'test1',
      updated: '2023-09-06 12:48:23.308Z'
    }
  },
  id: 'vfjjmyu7dn1cm7u',
  rel2: '5hrhujcfz6k6qon',
  text: 'test2',
  updated: '2023-09-06 13:22:02.940Z'
}
// record2
{
  collectionId: 'jdr3sz2uro70m1i',
  collectionName: 'testrel1',
  created: '2023-09-06 12:57:43.210Z',
  id: 'wu7sqtqzbz48z2g',
  text: 'test2',
  updated: '2023-09-06 12:57:43.210Z'
}
```


**output with multiple relation on same field rel2:**
```js
// record
{
  collectionId: 'p6fvms0nkje2n2t',
  collectionName: 'testrel2',
  created: '2023-09-06 12:48:16.962Z',
  expand: { rel2: [ [Object] ] },
  id: 'vfjjmyu7dn1cm7u',
  rel2: [ '5hrhujcfz6k6qon' ],
  text: 'test2',
  updated: '2023-09-06 13:55:55.493Z'
}
// record2
{
  collectionId: 'jdr3sz2uro70m1i',
  collectionName: 'testrel1',
  created: '2023-09-06 12:57:43.210Z',
  id: 'wu7sqtqzbz48z2g',
  text: 'test2',
  updated: '2023-09-06 12:57:43.210Z'
}
```

The initial output in both entries shows the id value of the related entry on `rel2`. However only on a single relation the `expand` shows further information. On the multiple relation `expand` shows even less, just displaying the type. 
Im sure it's just a notation thing but I would greatly appreciate clearing that up.

**Reply by ganigeorgiev** [reply]
> However only on a single relation the expand shows further information. On the multiple relation expand shows even less, just displaying the type.

I don't see why this is an issue. The record data is there under the `expand.rel2.{index}`.
If you don't see the record in the `console.log` output and just `[Object]` then this is normal.
At the end, the print/string object representation shouldn't really matter and it will differ depending on what is being printed and where the output is - in the browser console, node stdout/stderr, etc.

**Reply by Alexilator** [reply]
got it working, thanks!
